### PR TITLE
New feature flags backend

### PR DESCRIPTION
**Pull request by tiago-loureiro**
This PR intends to add an extra value to enable/disable SSO per team: current default is enabled. In addition, it moves all "flags" under a single CF since it's easier to manage and helps keeping all these "flags" in one place. Feel free to keep it as-is if you find that isolation is preferable.

TODO:
 - [x] Add integration tests for the new feature/endpoint
 - [x] Make spar check this endpoint
 - [x] Add endpoint in stern to check/set this value
 - ~~[ ] There's a TODO in the code, have a look at that too: separate PR~~

### CODE REVIEW

**Comment by fisx** [comment]
> [...] it moves all "flags" under a single CF since it's easier to manage and helps [...]

What's a "CF"?

**Comment by fisx** [comment]
> > [...] it moves all "flags" under a single CF since it's easier to manage and helps [...]
> 
> What's a "CF"?

ah.  column family.  very straight-forward.  :-)

**Comment by fisx** [comment]
>     * [ ]  There's a TODO in the code, have a look at that too.

this is about what should happen when sso is disabled.  spar needs to stop authenticating users against registered idps, but it shouldn't have to check on every authentication, just on idp registration.  options:

1. galley should notify spar if a team is disabled.  spar should keep the idp as registered, but with a local `enabled` flag that keeps it from using it for authentication.
2. galley should refuse to disable sso in teams that have registered idps.

2. is more vulnerable to race conditions (delete idp -> disable sso -> create a new one *just* fast enough to go before the last step -> now you have a disabled team with a working idp).  1. is probably a bit more work, and trickier to deploy with the required migrations.

i think i'll aim for 1.

File: services/galley/src/Galley/API.hs [hunk 1]
```diff
@@ -904,23 +906,49 @@ sitemap = do
...
+    put "/i/teams/:tid/features/sso" (continue Teams.setSSOStatusInternal) $
+        capture "tid"
+        .&. jsonRequest @SSOTeamConfig
+        .&. accept "application" "json"
+
+    -- End of team features
 
     get "/i/test/clients" (continue getClients)
         zauthUserId
+    -- TODO: What is this endpoint? Is this used anywhere?
```
Comment (jschaul) [inline services/galley/src/Galley/API.hs]: Grep or ag helps finding usage, it's used here: https://github.com/wireapp/wire-server/blob/develop/services/brig/test/integration/API/User/Client.hs#L319
Maybe you could add a comment and remove the TODO.

File: services/galley/test/integration/API/Teams.hs [hunk 2]
```diff
@@ -146,6 +150,27 @@ testCreateTeamWithMembers = do
...
+
+    let check :: HasCallStack => String -> SSOStatus -> TestM ()
+        check msg enabledness = do
+          SSOTeamConfig status <- jsonBody <$> (getSSOEnabledInternal tid <!! testResponse 200 Nothing)
+          liftIO $ assertEqual msg enabledness status
+
+    check "Teams should start with SSO disabled" SSODisabled
+
+    putSSOEnabledInternal tid SSOEnabled
+    check "Calling 'putEnabled True' should enable SSO" SSOEnabled
```
Comment (jschaul) [inline services/galley/test/integration/API/Teams.hs]: I like this, very clear.

File: services/galley/src/Galley/API.hs (hunk 1 above)
Comment (fisx) [reply services/galley/src/Galley/API.hs]: :+1:

(will do that in #817)

File: services/galley/test/integration/API/Teams.hs (hunk 2 above)
Comment (fisx) [reply services/galley/test/integration/API/Teams.hs]: thanks :)

**Pull request merged**